### PR TITLE
added clean architecture

### PR DESCRIPTION
**Pull request by PalmiraBenavente**
Added clean architecture, data layer, domain layer and app (presentation layer), also added a DI module for future dependency injections.
it was included in setting.gradle and added dependencies in the corresponding gladle

image

### CODE REVIEW

File: data/src/androidTest/java/com/example/data/ExampleInstrumentedTest.kt [hunk 1]
```diff
@@ -0,0 +1,24 @@
+package com.example.data
```
Comment (SartiMau) [inline data/src/androidTest/java/com/example/data/ExampleInstrumentedTest.kt]: you can delete this file

File: di/build.gradle [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        targetCompatibility JavaVersion.VERSION_1_8
+    }
+    kotlinOptions {
+        jvmTarget = '1.8'
+    }
+}
+
+dependencies {
+    implementation fileTree(dir: 'libs', include: ['*.jar'])
+    implementation project(path: ':domain')
```
Comment (SartiMau) [inline di/build.gradle:37]: DI module should know data too to be able to inject services and other data related things

File: di/src/androidTest/java/com/example/di/ExampleInstrumentedTest.kt [hunk 3]
```diff
@@ -0,0 +1,24 @@
+package com.example.di
```
Comment (SartiMau) [inline di/src/androidTest/java/com/example/di/ExampleInstrumentedTest.kt]: you can delete this too

File: domain/src/androidTest/java/com/example/domain/ExampleInstrumentedTest.kt [hunk 4]
```diff
@@ -0,0 +1,24 @@
+package com.example.domain
```
Comment (SartiMau) [inline domain/src/androidTest/java/com/example/domain/ExampleInstrumentedTest.kt]: same here

**Review by SartiMau — CHANGES_REQUESTED** [review]

**Review by LuchoElissondoGlobant — APPROVED** [review]
LG

File: data/src/androidTest/java/com/example/data/ExampleInstrumentedTest.kt (hunk 1 above)
Comment (PalmiraBenavente) [reply data/src/androidTest/java/com/example/data/ExampleInstrumentedTest.kt]: Done

File: di/build.gradle (hunk 2 above)
Comment (PalmiraBenavente) [reply di/build.gradle:37]: Done

File: di/src/androidTest/java/com/example/di/ExampleInstrumentedTest.kt (hunk 3 above)
Comment (PalmiraBenavente) [reply di/src/androidTest/java/com/example/di/ExampleInstrumentedTest.kt]: Done

File: domain/src/androidTest/java/com/example/domain/ExampleInstrumentedTest.kt (hunk 4 above)
Comment (PalmiraBenavente) [reply domain/src/androidTest/java/com/example/domain/ExampleInstrumentedTest.kt]: Done

**Review by nicolbidone — APPROVED** [review]

File: data/build.gradle [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    }
+    kotlinOptions {
+        jvmTarget = '1.8'
+    }
+}
+
+dependencies {
+    implementation fileTree(dir: 'libs', include: ['*.jar'])
+    implementation project(path: ':domain')
+    implementation project(path: ':di')
```
Comment (SartiMau) [inline data/build.gradle]: di should know data, not data di

File: di/src/androidTest/java/com/example/di/ExampleInstrumentedTest.kt [hunk 6]
```diff
@@ -0,0 +1,22 @@
+import androidx.test.platform.app.InstrumentationRegistry
```
Comment (SartiMau) [inline di/src/androidTest/java/com/example/di/ExampleInstrumentedTest.kt]: you can delete this file

**Review by SartiMau — CHANGES_REQUESTED** [review]

**Review by SartiMau — APPROVED** [review]

**Pull request merged**